### DISCUSSION TITLE
Support Pytorch quantization toolkit based QAT sample using FX backend of Torch-TRT

### DISCUSSION THREAD

**Discussion by peri044** [pytorch/TensorRT#1672] [RFCs]
# Pytorch quantization toolkit support in FX backend

## Goal(s)

Support Pytorch quantization toolkit based QAT sample using FX backend of Torch-TRT

## Proposed APIs / UX
 
The usage is similar to the sample here https://github.com/pytorch/TensorRT/blob/main/tests/py/qat/test_qat_trt_accuracy.py

### Limitations

No known limitations at this time

### Design
 

We need to write converters for quantize and dequantize ops in FX. Converters in TS : https://github.com/pytorch/TensorRT/blob/main/core/conversion/converters/impl/quantization.cpp

### Extensions Required to Core API implementations
 

N/A

#### Data Structures

N/A

### Details specific for TorchScript Support 

N/A

### Details specific for FX support 

Write new converters for quantization ops in FX. 

### Prototype - S
 

Implement a converters for quantization ops. Once this is done, use the QAT training sample and get a trained nn.Module and compile it with FX backend.

### MVP `(<1.4.0>)` - S
 

Implement a converters for quantization ops. Once this is done, use the QAT training sample and get a trained nn.Module and compile it with FX backend.

Both prototype and MVP would be the same for this feature. TS converters cannot be reused.

**Comment by peri044** [comment]
@narendasan  to review

**Comment by narendasan** [comment]
Seems good to me if its just a converter
